src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -653,6 +653,10 @@ class Config:
         if not new["enable"]:
             return
 
+        if new["certificate"]:
+            await self.middleware.call("certificate.cert_services_validation",
```

Comment:
This would be doing 2 things, 
1) It will raise verrors which I think we do not want and would only like after we have validated the bits down as well ? ( We can use verrors.extend here with the verrors being returned from the call )
2) Schema name being passed should be `activatedirectory_update.certificate`